rust/src/mark.rs

```diff
@@ -37,3 +37,24 @@ impl<'a> From<&'a am::marks::Mark<'a>> for Mark {
         }
     }
 }
+
+pub struct KeyValue {
+    pub key: String,
+    pub value: Value,
```

Comment:
```suggestion
    pub value: ScalarValue,
```

I'm reasonable certain that this should be `ScalarValue`, not the broader `Value` type, to align with the marks API updates. It used to return `Value`, but that was part of the refactoring bit - it's limited to `ScalarValue`.

Reply:
It would be nice, but this existing structure has a more general purpose. It is also used for `map_entries` where `Value` is mandatory.